model/task/task.go

```diff
@@ -2969,80 +2969,102 @@ func GetTasksByVersion(versionID string, opts GetTasksByVersionOptions) ([]Task,
 			if singleSort.Key == DisplayStatusKey || singleSort.Key == BaseTaskStatusKey {
-				sortAndPaginatePipeline = append(sortAndPaginatePipeline, addStatusColorSort((singleSort.Key)))
+				sortPipeline = append(sortPipeline, addStatusColorSort((singleSort.Key)))
 				sortFields = append(sortFields, bson.E{Key: "__" + singleSort.Key, Value: singleSort.Order})
 			} else {
 				sortFields = append(sortFields, bson.E{Key: singleSort.Key, Value: singleSort.Order})
 			}
 		}
-	}
-	sortFields = append(sortFields, bson.E{Key: IdKey, Value: 1})
+		sortFields = append(sortFields, bson.E{Key: IdKey, Value: 1})
 
-	sortAndPaginatePipeline = append(sortAndPaginatePipeline, bson.M{
-		"$sort": sortFields,
-	})
-
-	if opts.Limit > 0 {
-		sortAndPaginatePipeline = append(sortAndPaginatePipeline, bson.M{
-			"$skip": opts.Page * opts.Limit,
-		})
-		sortAndPaginatePipeline = append(sortAndPaginatePipeline, bson.M{
-			"$limit": opts.Limit,
+		sortPipeline = append(sortPipeline, bson.M{
+			"$sort": sortFields,
 		})
+
+		pipeline = append(pipeline, sortPipeline...)
 	}
+
 	if len(opts.FieldsToProject) > 0 {
 		fieldKeys := bson.M{}
 		for _, field := range opts.FieldsToProject {
 			fieldKeys[field] = 1
 		}
-		sortAndPaginatePipeline = append(sortAndPaginatePipeline, bson.M{
+		pipeline = append(pipeline, bson.M{
 			"$project": fieldKeys,
 		})
 	}
 
-	// Use a $facet to perform separate aggregations for $count and to sort and paginate the results in the same query
-	tasksAndCountPipeline := bson.M{
-		"$facet": bson.M{
-			"count": []bson.M{
-				{"$count": "count"},
+	// If there is a limit we should calculate the total count before we apply the limit and pagination
+	if opts.Limit > 0 {
+		paginatePipeline := []bson.M{}
+		paginatePipeline = append(paginatePipeline, bson.M{
+			"$skip": opts.Page * opts.Limit,
+		})
+		paginatePipeline = append(paginatePipeline, bson.M{
+			"$limit": opts.Limit,
+		})
+		// Use a $facet to perform separate aggregations for $count and to sort and paginate the results in the same query
+		tasksAndCountPipeline := bson.M{
+			"$facet": bson.M{
+				"count": []bson.M{
+					{"$count": "count"},
+				},
+				"tasks": paginatePipeline,
 			},
-			"tasks": sortAndPaginatePipeline,
-		},
+		}
+		pipeline = append(pipeline, tasksAndCountPipeline)
 	}
 
-	pipeline = append(pipeline, tasksAndCountPipeline)
-
-	type TasksAndCount struct {
-		Tasks []Task           `bson:"tasks"`
-		Count []map[string]int `bson:"count"`
-	}
-	results := []TasksAndCount{}
 	env := evergreen.GetEnvironment()
 	ctx, cancel := env.Context()
 	defer cancel()
 	cursor, err := env.DB().Collection(Collection).Aggregate(ctx, pipeline)
 	if err != nil {
 		return nil, 0, err
 	}
-	err = cursor.All(ctx, &results)
-	if err != nil {
-		return nil, 0, err
+
+	type TasksAndCount struct {
+		Tasks []Task           `bson:"tasks"`
+		Count []map[string]int `bson:"count"`
+	}
+	TaskAndCountResults := []TasksAndCount{}
+	TaskResults := []Task{}
```

Comment:
Can just define this in these variables  in the if/else statements they're used in